### PR TITLE
Fix decoding of null choice elements

### LINKED ISSUE #27: Empty struct decoding
Add tests and debug the empty struct scenario (e.g., `<br />`)

### PR DESCRIPTION
**Pull request by bwetherfield**
Wraps null elements in a `ChoiceBox` so that they get picked up in the `XMLChoiceKey` case and can  be unwrapped to reveal a `NullBox` otherwise. 
Fixes #27

### CODE REVIEW

File: Tests/XMLCoderTests/SimpleChoiceTests.swift [hunk 1]
```diff
@@ -31,7 +31,6 @@ extension IntOrString: Codable {
 
     init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        print(type(of: container))
```
Comment (jsbean) [inline Tests/XMLCoderTests/SimpleChoiceTests.swift:34]: 🙌🏻

File: Sources/XMLCoder/Decoder/XMLDecoderImplementation.swift [hunk 2]
```diff
@@ -411,10 +419,10 @@ extension XMLDecoderImplementation {
     }
 
     func unbox<T: Decodable>(_ box: Box) throws -> T {
-
+
```
Comment (jsbean) [inline Sources/XMLCoder/Decoder/XMLDecoderImplementation.swift]: Before we take this in, can we get delete the whitespace changes? 🥶

Comment (bwetherfield) [reply Sources/XMLCoder/Decoder/XMLDecoderImplementation.swift]: :grimacing: whoops sorry
